index.js

```diff
@@ -116,22 +116,61 @@ function processRequestResponse(options, error, response, body) {
   var callback = options.callback;
 
   var stringBody;
+  var isTargetPage;
   var isChallengePresent;
   var isRedirectChallengePresent;
 
+  // Encoding is null so body should be a buffer object
   if (error || !body || !body.toString) {
     // Pure request error (bad connection, wrong url, etc)
     error = new errors.RequestError(error, options, response);
 
     return callback(error, response, body);
   }
 
-  stringBody = body.toString('utf8');
+  isTargetPage = response.statusCode < 500 ||
+      !(response.caseless.has('server') &&
+        /^cloudflare/i.test(response.caseless.get('server')));
+
+  if (!isTargetPage) {
+    if (body.length < 1) {
+      // This is a cloudflare response, 5xx with an empty body.
+      if (response.request.METHOD !== 'HEAD') {
+        error = new errors.CloudflareError(response.statusCode, options, response);
```

Comment:
https://support.cloudflare.com/hc/en-us/articles/115003011431-5xx-Server-Error

Reply:
The validate function will throw for `<span class="cf-error-code">5xx</span>` as well as `<span class="cf-error-code">1xxx</span>`.

Reply:
Updated link, the links in test files are dead: https://support.cloudflare.com/hc/en-us/sections/200820298-Error-Pages

Reply:
It would be nice if we added messages to theses errors:
> Error 1013 HTTP hostname and TLS SNI hostname mismatch
> Error 1001: DNS resolution error
> Error 1000: DNS points to prohibited IP
> Error 1002: DNS points to Prohibited IP
> Error 1002: Restricted
> Error 1003 Access Denied: Direct IP Access Not Allowed
> Error 1004: Host Not Configured to Serve Web Traffic
> Error 1006, 1007 & 1008 Access Denied: Your IP address has been banned
> Error 1010: The owner of this website has banned your access based on your browser's signature
> 1011 Error: Access Denied (Hotlinking Denied)
> 1012 Error: Access Denied
> Error 1016 - Origin DNS error